app/dashboard/models.py

```diff
@@ -721,7 +721,7 @@ def status(self):
         is_traditional_bounty_type = self.project_type == 'traditional'
         try:
             has_tips = self.tips.filter(is_for_bounty_fulfiller=False).send_happy_path().exists()
-            if has_tips and is_traditional_bounty_type:
+            if has_tips and is_traditional_bounty_type and self.idx_status in ['cancelled'] :
```

Comment:
this will create a circular logic where the `bounty.status` is toggled back and forth between `done` and `cancelled`, wont it?  the bounty.status will be marked `done` then `cancelled` and back and forth forever.

Reply:
note dashboard.models:psave_bounty where

```
instance.idx_status = instance.status
``` 

is set

Reply:
> this will create a circular logic where the bounty.status is toggled back and forth between done and cancelled, won't it ?

Could you give an example?

bounty.status depends on the bounty.idx_status + business logic to decide the value.
If the idx_status is done -> the status would be set to done via https://github.com/gitcoinco/web/pull/5447/files#diff-0d2d86c824ef942c5a3a01c83ddfe758R728 right ?


Another solution which we could do make it even better:

>  `and self.idx_status in ['cancelled'] and not self.is_open`

Reply:
im not sure how to give you an example; except to say that since you guys merged this, instance.idx_status is set by instance.status, and then instance.idx_status is set by instance.idx_status.

see line 1669 of dashboard/models.py to see what i mean.

this is circular dependancy and will cause the idx_status to toggle back and forth between the two statuses each time idx_status is computed.

Reply:
here
```
In [11]: from dashboard.models import Bounty
    ...: bounty = Bounty.objects.last()
    ...: bounty.idx_status = 'cancelled'
    ...: bounty.save()
    ...: print(bounty.idx_status)
    ...: bounty.save()
    ...: print(bounty.idx_status)
    ...:
done
open

In [12]: from dashboard.models import Bounty
    ...: bounty = Bounty.objects.last()
    ...: bounty.idx_status = 'cancelled'
    ...: bounty.save()
    ...: print(bounty.idx_status)
    ...: bounty.save()
    ...: print(bounty.idx_status)
    ...:
done
open

```